google/cloud/spanner/integration_tests/database_admin_integration_test.cc

```diff
@@ -461,16 +464,21 @@ TEST_F(DatabaseAdminClientTest, CreateWithEncryptionKey) {
 // @test Verify creating a database fails if a nonexistent encryption key is
 // supplied.
 TEST_F(DatabaseAdminClientTest, CreateWithNonexistentEncryptionKey) {
-  if (emulator_) GTEST_SKIP() << "emulator does not support CMEK";
   KmsKeyName nonexistent_encryption_key(instance_.project_id(), location_,
                                         kKeyRing, "ceci-n-est-pas-une-cle");
   auto database =
       client_
           .CreateDatabase(database_, /*extra_statements=*/{},
                           CustomerManagedEncryption(nonexistent_encryption_key))
           .get();
-  EXPECT_THAT(database, StatusIs(StatusCode::kFailedPrecondition,
-                                 HasSubstr("KMS Key provided is not usable")));
+  if (emulator_) {
+    EXPECT_THAT(database, IsOk());
+    EXPECT_STATUS_OK(client_.DropDatabase(database_));
+  } else {
+    EXPECT_THAT(database,
+                StatusIs(StatusCode::kFailedPrecondition,
+                         HasSubstr("KMS Key provided is not usable")));
+  }
```

Comment:
This is not testing the library, it is testing the service.  As long as the library passes the right request, I am not sure why we care if the service returns an error or not.

Reply:
One reason to look at the result (also not related to testing the library) is that we should drop the database if it succeeds.

But, in the big picture, I certainly agree with you.  The whole test, as in other such examples, was probably the product of a "thou shalt have an integration test for unusable keys" directive from the Spanner client libraries team, and my unwillingness to argue with them (much).